src/storage.ts

```diff
@@ -388,9 +389,15 @@ export class Storage extends Service {
     options = Object.assign({}, options, {
       apiEndpoint: options.apiEndpoint || 'storage.googleapis.com',
     });
-    const url =
-      process.env.STORAGE_EMULATOR_HOST ||
-      `https://${options.apiEndpoint}/storage/v1`;
+
+    // Overrides options.apiEndpoint.
+    const EMULATOR_HOST = process.env.STORAGE_EMULATOR_HOST;
+    if (typeof EMULATOR_HOST === 'string') {
+      options.apiEndpoint = new URL(EMULATOR_HOST).host;
```

Comment:
Maybe we should just accept raw user-input-- the whole thing, protocol, port, and path. We can also append "/storage/v1" to it to take the path of least surprise for the user (and less cognitive overload for us).